integration-tests/src/test/java/org/apache/druid/tests/query/ITWikipediaQueryTest.java

```diff
@@ -85,37 +88,54 @@ public void testWikipediaQueriesFromFile() throws Exception
   @Test
   public void testQueryLaningLaneIsLimited() throws Exception
   {
-    // the broker is configured with a manually defined query lane, 'one' with limit 1
-    //  -Ddruid.query.scheduler.laning.type=manual
-    //  -Ddruid.query.scheduler.laning.lanes.one=1
-    // by issuing 50 queries, at least 1 of them will succeed on 'one', and at least 1 of them will overlap enough to
-    // get limited
-    final int numQueries = 50;
-    List<Future<StatusResponseHolder>> futures = new ArrayList<>(numQueries);
-    for (int i = 0; i < numQueries; i++) {
-      futures.add(
-          queryClient.queryAsync(
-              queryHelper.getQueryURL(config.getBrokerUrl()),
-              getQueryBuilder().build()
-          )
-      );
-    }
-
-    int success = 0;
-    int limited = 0;
-
-    for (Future<StatusResponseHolder> future : futures) {
-      StatusResponseHolder status = future.get();
-      if (status.getStatus().getCode() == QueryCapacityExceededException.STATUS_CODE) {
-        limited++;
-        Assert.assertTrue(status.getContent().contains(QueryCapacityExceededException.makeLaneErrorMessage("one", 1)));
-      } else if (status.getStatus().getCode() == HttpResponseStatus.OK.getCode()) {
-        success++;
-      }
-    }
-
-    Assert.assertTrue(success > 0);
-    Assert.assertTrue(limited > 0);
+    ITRetryUtil.retryUntil(
+        () -> {
+          // the broker is configured with a manually defined query lane, 'one' with limit 1
+          //  -Ddruid.query.scheduler.laning.type=manual
+          //  -Ddruid.query.scheduler.laning.lanes.one=1
+          // by issuing 50 queries, at least 1 of them will succeed on 'one', and at least 1 of them will overlap enough to
+          // get limited.
```

Comment:
Have you considered introducing an artificial sleep on the query execution path for queries issued by this test? 

I think we can accomplish this with what we have today by using one of the javascript features (aggregators, filters, etc.) available to us.

This way we can guarantee overlaps, instead of having to rely on a re-try mechanism?

Reply:
That's an interesting idea, I searched a bit for how to sleep in javascript, but our JS engine doesn't seem to support "Atomics.wait" and the filter/aggregator functions don't seem to run in an `async` JS function, so we can't wait on a promise either.

Reply:
ah too bad. TIL javascript has no sleep function 🤯